src/charm.py

```diff
@@ -224,10 +224,16 @@ def _check_precondition(self) -> None:
             )
         nginx_route_relation = self.model.get_relation("nginx-route")
         ingress_relation = self.model.get_relation("ingress")
+        # The charm only supports one relation at a time, so we check if both are present
+        # and raise an error if they are.
         if nginx_route_relation is not None and ingress_relation is not None:
             raise InvalidIngressError(
                 "nginx-ingress-integrator cannot establish more than one relation at a time"
             )
+        hostnames = self.get_additional_hostnames()
+        if ingress_relation is not None and len(hostnames) > 1:
+            self._ingress_provider.wipe_ingress_data(ingress_relation)
```

Comment:
We're adding the `wipe_ingress_data` here and in `generate_ingress_url` when an `InvalidIngressError` is raised.
I'm wondering if we should have a more generic approach and call the wipe where we catch the `InvalidIngressError` (l336) ?

Reply:
if we do this, it will require a lot of extra logic since `relation` will be unbound, etc. we will have to have logic to check that the relation is set, that we are using ingress relation etc. this will include repetition of calling potentially unsafe in an `except` branch

Reply:
Ok let's keep it as it is, but I'm wondering if we shouldn't systematically `wipe_ingress_data` when we have an `InvalidIngressError` whatever the issue is.